src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,63 @@
+package step2;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String DEFAULT_DELIMITER = ",|:";
+    public static final String CHARACTER_OR_NEGATIVE_NUMBER_ERROR = "문자와 음수는 입력이 불가능 합니다.";
+
+    private StringAddCalculator() {
+    }
+
+    public static int splitAndSum(String input) {
+
+        if (input == null || input.isEmpty()) {
+            return 0;
+        }
+
+        String delimiter = DEFAULT_DELIMITER;
+        String beforeSplit = input;
+        Matcher matcher = Pattern.compile("//(.)\n(.*)").matcher(input);
+
+        if (matcher.find()) {
+            delimiter = matcher.group(1);
+            beforeSplit = matcher.group(2);
+        }
+
+        String[] splitNumbers = splitDelimiter(beforeSplit, delimiter);
+        Arrays.stream(splitNumbers).forEachOrdered(StringAddCalculator::isDigit);
+
+        return add(Arrays.stream(splitNumbers)
+                .mapToInt(Integer::parseInt)
+                .toArray());
```

Comment:
```suggestion
        return Arrays.stream(splitDelimiter(beforeSplit, delimiter))
                .mapToObj(PositiveNumber::valueOf)
                .reduce(0, PositiveNumber::add);
```
이런식으로 표현할 수 있도록 `PositiveNumber`라는 원시값 포장 객체를 만들어보면 어떨까요?
검증 로직도 PositiveNumber에게 위임하면 될 것 같습니다.

Reply:
PositiveNumber 객체 생성 하였습니다.
결과 반환은 구현이 말씀하신거와 조금 달라서 Stream 없이 진행 하였습니다.